File: packages/amplify_core/lib/src/types/analytics/analytics/analytics_event.dart
// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
// SPDX-License-Identifier: Apache-2.0

import 'package:amplify_core/src/types/analytics/analytics/custom_properties.dart';

class AnalyticsEvent {
  AnalyticsEvent(this.name);

  final String name;
  final CustomProperties customProperties = CustomProperties();
}

File: packages/amplify_datastore/ios/Internal/Amplify/Categories/Analytics/Event/AnalyticsEvent.swift
//
// Copyright Amazon.com Inc. or its affiliates.
// All Rights Reserved.
//
// SPDX-License-Identifier: Apache-2.0
//

import Foundation

/// Analytics event
public protocol AnalyticsEvent {

    /// Name of the event
    var name: String { get }

    // Properties of the event
    var properties: AnalyticsProperties? { get }
}
